### PR TITLE
44 proposal3 -- Ready for Merge

### PR DESCRIPTION
**Pull request by lc9900**
* implemented updateMeetup method to check for conflict when meetup is being update. If found, error is thrown, and no update will be done.
* implemented adding meetup with necessary error -- schedule in the past error, scheduling conflict error.
* State.user now has friends as well.
* Fixed a bug where on refresh, state.user loses `places`

Something's going on with Google login, it doesn't work now. Seems to be due to our google account's api issue. I switched to the old branch where I implemented google login, and that doesn't work neither.
Will look into this some other time.

### CODE REVIEW

File: client/components/NewMeetup.js [hunk 1]
```diff
@@ -2,20 +2,164 @@ import React, {Component} from 'react';
...
 import { Redirect } from 'react-router-dom';
+import * as _ from 'lodash';
+import axios from 'axios';
+import {daysInMonth} from '../../utils';
 
 class NewMeetup extends Component {
-    constructor(){
+    constructor(props){
         super();
+        let { user } = props,
```
Comment (ajfrank) [inline client/components/NewMeetup.js]: Why are you using a "let" here instead of a "const"?

File: client/components/NewMeetup.js [hunk 2]
```diff
@@ -2,20 +2,164 @@ import React, {Component} from 'react';
...
+            now = new Date(),
```
Comment (ajfrank) [inline client/components/NewMeetup.js]: Do you need to store each of these values in a variable this way or is there some way to grab everything in one variable and unpack as needed?

File: client/components/NewMeetup.js [hunk 3]
```diff
@@ -2,20 +2,164 @@ import React, {Component} from 'react';
...
+          input_err: '',
+          input_success: ''
+        };
+        this.handleSubmit = this.handleSubmit.bind(this);
+        this.handleChange = this.handleChange.bind(this);
+    }
+
+    handleSubmit(event) {
+      event.preventDefault();
+      let {input_year, input_month, input_date, input_hour, input_friend} = this.state;
```
Comment (ajfrank) [inline client/components/NewMeetup.js]: Same question as above - any reason for the "let"?

File: client/components/NewMeetup.js [hunk 4]
```diff
@@ -2,20 +2,164 @@ import React, {Component} from 'react';
...
+      const {user} = this.props;
+      // console.log(this.state);
+      axios.post(`/api/meetup/add/${user.id}`, {
```
Comment (ajfrank) [inline client/components/NewMeetup.js]: Do you want to do the axios call here or would it be better to do it in a thunk?

File: client/components/NewMeetup.js (hunk 1 above)
Comment (lc9900) [reply client/components/NewMeetup.js]: just being lazy =).
didn't want to type const {user} = props;
then let now = new Date();
etc. =)
but in all fairness you are right. I'll change it.

File: client/components/NewMeetup.js (hunk 3 above)
Comment (lc9900) [reply client/components/NewMeetup.js]: You are right, I should use const

File: client/components/NewMeetup.js (hunk 4 above)
Comment (lc9900) [reply client/components/NewMeetup.js]: I was debating about that. But those doesn't change any redux state, so no dispatch. Feel like an overkill to create all the reducer files just to add a thunk that doesn't have anything to do with redux state.

File: client/components/NewMeetup.js (hunk 2 above)
Comment (lc9900) [reply client/components/NewMeetup.js]: just need now_year, and now_month. Going to take out now_date, and now_hour.
It's more for readability.

**Pull request merged**